tests/Feature/Components/Progress/IndexTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+it('can render')
+    ->expect('<x-progress percent="35" />')
+    ->render()
+    ->toContain('35%')
+    ->toContain('bg-primary-600');
+
+it('can render size variations', function (array $size) {
+    $key = array_key_first($size);
+    $class = $size[$key];
+
+    $component = <<<'HTML'
+    <x-progress percent="35" {{ size }} />
+    HTML;
+
+    $component = str_replace('{{ size }}', $key, $component);
+
+    expect($component)->render()
+        ->toContain('35%')
+        ->toContain($class);
+})->with([
+    fn () => ['xs' => 'h-2.5'],
+    fn () => ['sm' => 'h-3'],
+    fn () => ['md' => 'h-4'],
+    fn () => ['lg' => 'h-5'],
+]);
+
+it('can render red color')
```

Comment:
Use the data providers `colors` to test all colors. Check for other examples that are using this data provider.